parsl/executors/workqueue/executor.py

```diff
@@ -333,25 +333,25 @@ def submit(self, func, resource_specification, *args, **kwargs):
         cores = None
         memory = None
         disk = None
+        gpus = None
         if resource_specification and isinstance(resource_specification, dict):
             logger.debug("Got resource specification: {}".format(resource_specification))
 
-            acceptable_resource_types = set(['cores', 'memory', 'disk'])
+            acceptable_resource_types = set(['cores', 'memory', 'disk', 'gpus'])
             keys = set(resource_specification.keys())
             if self.autolabel:
                 if not keys.issubset(acceptable_resource_types):
-                    logger.error("Task resource specification only accepts "
-                                 "three types of resources: cores, memory, and disk")
-                    raise ExecutorError(self, "Task resource specification only accepts "
-                                              "three types of resources: cores, memory, and disk")
+                    message = "Task resource specification only accepts these types of resources: {}".format(
+                            ', '.join(acceptable_resource_types))
+                    logger.error(message)
+                    raise ExecutorError(self, message)
 
             elif keys != acceptable_resource_types:
                 logger.error("Running with `autolabel=False`. In this mode, "
                              "task resource specification requires "
                              "three resources to be specified simultaneously: cores, memory, and disk")
                 raise ExecutorError(self, "Task resource specification requires "
-                                          "three resources to be specified simultaneously: cores, memory, and disk, "
-                                          "and only takes these three resource types, when running with `autolabel=False`. "
+                                          "three resources to be specified simultaneously: cores, memory, and disk. "
```

Comment:
this branch of the `if` fires when one of the (now) four resource types is missing, I think?

Reply:
You're right, I redid those checks.

Also, WQ will be changing its requirements in the near future, so we'll be able to drop the "all 3 required" rule then